packages/provider/index.js

```diff
@@ -4,6 +4,8 @@ const { createInterfaceAdapter } = require("@truffle/interface-adapter");
 const wrapper = require("./wrapper");
 const DEFAULT_NETWORK_CHECK_TIMEOUT = 5000;
 
+let id = 0;
```

Comment:
I don't think the `id` should be a module singleton. We should probably declare it in the scope in which we add the `.request` method onto the `provider`.

The issue with the singleton is that a) it is only used if we add the `request` method to the `provider` ourselves. b) it makes for potentially non-deterministic JSON-RPC ids if `packages/provider` is ever used multiple times in a single node session. The second point likely isn't an issue with typical Truffle usage today, since Truffle (and the provider package) isn't used as a user library, but it's likely worth thinking about.

Reply:
Thanks for the explanation, @davidmurdoch . I agree that the `id` should be declared within the scope of the method.